mindsdb/integrations/handlers/huggingface_handler/huggingface_handler.py

```diff
@@ -0,0 +1,137 @@
+import pandas as pd
+import transformers
+
+from mindsdb.integrations.libs.base import BaseMLEngine
+
+
+class HuggingFaceHandler(BaseMLEngine):
+    name = 'huggingface'
+
+    def create(self, target, args=None, **kwargs):
+
+        args['target'] = target
+
+        model_name = args['model_name']
+        hf_model_storage_path = self.engine_storage.folder_get(model_name)  # real
```

Comment:
hey, I am a little confused, why use engine_storage to store the model? why not use path generated by self.model_storage?

Reply:
Hey! @ea-rus can expand further, but one of the main reasons is that some MindsDB installations (like the cloud) may have multiple machines that will want to access (read or write) a model's artifacts. For that, we need a mechanism that will guarantee access to all, even if the storage is on instances that are completely separated from the rest of the architecture. Using "just the path" may be too brittle a solution.

Reply:
Also difference between engine and model storage is scoupe:
- model_storage is used only by one model
- engine_storage is global for engine. We are using it here to download huggingface model once and store it globaly to not download it for every new model